ingestion-beam/src/main/java/com/mozilla/telemetry/amplitude/SendRequest.java

```diff
@@ -0,0 +1,169 @@
+package com.mozilla.telemetry.amplitude;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.mozilla.telemetry.metrics.KeyedCounter;
+import com.mozilla.telemetry.transforms.FailureMessage;
+import com.mozilla.telemetry.util.Json;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Map;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.metrics.Distribution;
+import org.apache.beam.sdk.metrics.Metrics;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.WithFailures.ExceptionElement;
+import org.apache.beam.sdk.transforms.WithFailures.Result;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+/**
+ * Send GET requests of event batches to Amplitude endpoint.
+ */
+@SuppressWarnings("checkstyle:lineLength")
+public class SendRequest extends
+    PTransform<PCollection<KV<String, Iterable<AmplitudeEvent>>>, Result<PCollection<KV<String, Iterable<AmplitudeEvent>>>, PubsubMessage>> {
+
+  private static OkHttpClient httpClient;
+
+  private final Distribution requestTimer = Metrics.distribution(SendRequest.class,
+      "reporting_request_millis");
+  private final Boolean reportingEnabled;
+  private final String amplitudeUrl;
+  private final Integer maxBatchesPerSecond;
+  private final Map<String, String> apiKeys;
+
+  public static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+
+  private static class HttpRequestException extends IOException {
+
+    HttpRequestException(String message, int code) {
+      super("HTTP " + code + ": " + message);
+    }
+  }
+
+  private static class HttpRedirectException extends IOException {
+
+    HttpRedirectException(String url) {
+      super("Redirects not allowed: " + url);
+    }
+  }
+
+  /**
+   * Constructor without custom Amplitude URL.
+   */
+  public SendRequest(Map<String, String> apiKeys, Boolean reportingEnabled,
+      Integer maxBatchesPerSecond) {
+    this.reportingEnabled = reportingEnabled;
+    this.amplitudeUrl = "https://api2.amplitude.com";
+    this.maxBatchesPerSecond = maxBatchesPerSecond;
+    this.apiKeys = apiKeys;
```

Comment:
I think this is also possible
```suggestion
    this(apiKeys, reportingEnabled, maxBatchesPerSecond, "https://api2.amplitude.com");
```